docs/diagrams/ModelClassDiagram.puml

```diff
@@ -68,7 +73,7 @@ Name -[hidden]right-> DateOfBirth
 DateOfBirth -[hidden]right-> Gender
 Species -[hidden]right-> Food
 
-PetTracker *--> "1" Schedule
```

Comment:
Your composition and aggregation seems to be messed up.
I don't think UniquePetList's relationship with Schedule is composition.
And Slot and PetAmountAndPet's relationship with Pet is not composition either.
I'm not sure about the rest but you should check them too.

Reply:
I just fixed the errors hahaha. Can you glance through again? Thx!

Reply:
PetAmountAndPet looks like an association class to me.
I don't its relationship with pet is "has a".

Reply:
Hmmmm kinda makes sense. Only doubt I have is that my FoodAmountAndPet(realized there's a type here. gonna change later) doesn't know FoodCollection. Would this violate the point of Association class? The way I see it was more based on the fact that FoodCollection has a whole-part relationship with FoodAmountAndPet, hence the implementation.

Reply:
Why is there no arrow between FoodCollection and Food?
Maybe we can standardize whether to use Integer or int?
FoodAmountAndPet contains Pet? I can't process.

Reply:
Actually I'm able to remove the association between pet and FoodAmountAndPet. Would this make more sense?

Reply:
Haven't started revising yet so I'm not really sure but intuitively FoodAmountAndPet feels like an association class between FoodAmount and Pet.
https://stackoverflow.com/questions/37767328/implementation-of-classes-with-association-in-java
An association class would be able to have both as variables. However, FoodAmount is just an integer here.